docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+To help with abiding by the rules of commit messages, please use the `commitizen` tool mentioned in the documentation above. This means will we use `git cz` instead of `git commit`.
+
+In my case, I want the commit message to look like this:
+
+```
+docs: tutorial on how to submit a pull request to lb4
+
+This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to Loopback v4
+
+```
+
+If you need the long description to span multiple lines, use `\n` characters to separate the sentences.
+
+Let's go through the interactive prompts of `commitizen`.
+
+Type
+```
+git cz
+```
+
+The first prompt asks you to choose the `type` value:
+
+![submit_pr_git_cz_1.png](./imgs/submit_pr_git_cz_1.png)
+
+
+Using the `down arrow` key, I will choose `docs` and press `ENTER`.
+
+![submit_pr_git_cz_2.png](./imgs/submit_pr_git_cz_2.png)
+
+The next prompt asks you to choose the `scope` value:
+
+![submit_pr_git_cz_3.png](./imgs/submit_pr_git_cz_3.png)
+
+In my case, I won't be specifying a scope value, so I will simply press `ENTER` to skip.
+
+The next prompt asks you to specify the `short description` value:
+
+![submit_pr_git_cz_4.png](./imgs/submit_pr_git_cz_4.png)
+
+I specify : `tutorial on how to submit a pull request to lb4` .
+
+![submit_pr_git_cz_5.png](./imgs/submit_pr_git_cz_5.png)
+
+The next prompt asks you to specify the `long description` value:
+
+![submit_pr_git_cz_6.png](./imgs/submit_pr_git_cz_6.png)
+
+I specify : `This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to Loopback v4`
+
+![submit_pr_git_cz_7.png](./imgs/submit_pr_git_cz_7.png)
+
+
+The next prompt asks if any `breaking changes` will be introduced.
+
+![submit_pr_git_cz_8.png](./imgs/submit_pr_git_cz_8.png)
+
+I specify : `N`
+
+![submit_pr_git_cz_9.png](./imgs/submit_pr_git_cz_9.png)
+
+The next prompt asks if this change affects any open issues.
+
+![submit_pr_git_cz_10.png](./imgs/submit_pr_git_cz_10.png)
+
+I specify : `N`
+
+![submit_pr_git_cz_11.png](./imgs/submit_pr_git_cz_11.png)
+
+The interactive `commitizen` prompts complete and the commit is created with a properly formatted message.
+
+![submit_pr_git_cz_12.png](./imgs/submit_pr_git_cz_12.png)
+
+To view the commit message that `commitizen` has created, you can type :
+
+```
+git log
+```
+
+![submit_pr_git_cz_13.png](./imgs/submit_pr_git_cz_13.png)
+
+Press `q` to exit the list of commits.
+
+
+### 14. Push commits to your remote feature branch
+
+It is now time to push your local committed changes from your local feature branch directory to your remote feature branch; to keep them in synch.
+
+Type :
+
+```
+git push
+```
+
+![submit_pr_git_push_1.png](./imgs/submit_pr_git_push_1.png)
+
+Open your browser, and navigate to your forked repository, and you will notice the commit.
+
+![submit_pr_git_push_2.png](./imgs/submit_pr_git_push_2.png)
+
+
+### 15. Rebasing
```

Comment:
We should add a section to describe how to add more commits to a given PR. New users are often confused as they don't know the PR automatically picks up the latest version in your branch against the base. There is no need to close the current PR and reopen a new one.

We also should describe what's best practice to add commits to a PR under review.

Reply:
I actually have  a question about commits "before a PR" and "after a PR". 

When someone creates a PR for the first time, is it expected that there is 1 commit with a valid `commitizen` message? Or does this only matter ONLY before the maintainers merge the PR?

Same question after the PR is created (let's say with 1 commit and valid commitizen message).  During the review process, when many commits are added, do they all have to have valid `commitizen` messages, or does this only matter ONLY before the maintainers merge the PR?

I asked a few weeks ago, and I was told that every commit message should be a valid `commitizen` message.

Reply:
@raymondfeng , I spoke with @b-admike this morning to address my questions above. My latest commit discusses the number of commits before creating a PR, and when it is about to be merged by maintainers.